### PR TITLE
PINF-809 Dual-host ingress & TLS SAN templating (astronomer umbrella chart)

### PR DESCRIPTION
**Pull request by adhipjos**
## Description

Implements the **astronomer umbrella chart** half of dual-host ingress & TLS SAN templating for Control Plane HA (PINF-809).

When `global.controlPlaneHA.enabled` is set, each customer-facing ingress (public, houston, grafana, alertmanager, prometheus) emits **two host families**:

- **per-CP admin**: the existing host from `global.baseDomain` (unchanged)
- **global customer-facing**: a new host from `global.controlPlaneHA.globalBaseDomain`

The `spec.tls[].hosts` list mirrors both families on each ingress, so each per-CP cert's SAN expectation lines up with the hostnames it serves. With HA off, rendering is unchanged (single per-CP host family only).

Details:
- **public** (`charts/astronomer/templates/ingress.yaml`): emits `<globalBaseDomain>` + `app.<globalBaseDomain>` (both → astro-ui), extends the bare-domain→`app.` redirect snippet to the global host, mirrors the two global hosts into TLS hosts.
- **houston / grafana / alertmanager / prometheus**: add a second `host:` rule + TLS host; grafana/alertmanager/prometheus gain a `*.globalUrl` named helper next to the existing `*.url`.
- **prometheus**: the global host is gated by `$emitGlobalHost` (HA **and** control/unified plane) because `globalBaseDomain` is control-plane-only and unset on data planes with shared values.
- **docs**: abbreviated per-CP cert SAN directive added as comments in `values.yaml` under `global.controlPlaneHA.globalBaseDomain` (both host families, one physical cert per CP, DNS-01 ACME). Full customer-facing doc tracked in PINF-813.

Cert issuance itself (DNS-01 ACME) is a customer prerequisite. The chart only expands the ingress `tls.hosts` list.

> **Depends on PINF-768 (#3302)**, now merged into `feature/cp-ha-dr`, which introduces the `global.controlPlaneHA.*` values this consumes. This PR targets `feature/cp-ha-dr` and contains only the PINF-809 ingress/TLS changes.

## Related Issues

Related astronomer/issues#PINF-809
Consumes values from PINF-768 (#3302). Full SAN documentation tracked in PINF-813.

## Testing

- `uv run pytest tests/chart_tests/test_dual_host_ingress.py`: **60 passed** (HA-off single family, HA-on dual family + TLS SAN expansion across all 5 ingresses, prometheus data-plane regression guard, public bare-domain redirect).
- `uv run pytest tests/chart_tests/test_control_plane_ha.py`: 95 passed (no PINF-768 regressions).
- `uv run pytest tests/chart_tests/ -n auto`: **4667 passed, 72 skipped** (full suite, no regressions).
- Manual `helm template` renders verified for HA-on, HA-off, and the prometheus data-plane edge case.

## Merging

Target: `feature/cp-ha-dr` (CP-HA integration branch). Part of the 2.1.0 release line. No cherry-picks to older release branches.

### CODE REVIEW

File: charts/astronomer/templates/ingress.yaml [hunk 1]
```diff
@@ -27,6 +27,11 @@ metadata:
       if ($host = '{{ .Values.global.baseDomain }}' ) {
         rewrite ^ https://app.{{ .Values.global.baseDomain }}$request_uri permanent;
       }
+      {{- if .Values.global.controlPlaneHA.enabled }}
+      if ($host = '{{ .Values.global.controlPlaneHA.globalBaseDomain }}' ) {
+        rewrite ^ https://app.{{ .Values.global.controlPlaneHA.globalBaseDomain }}$request_uri permanent;
+      }
+      {{- end }}
```
[the remark this answers is not in the document]
Comment (adhipjos) [reply charts/astronomer/templates/ingress.yaml]: Fixed in b37b6657b. Introduced $emitGlobalHost (controlPlaneHA.enabled AND control/unified plane, matching the prometheus ingress) and gated the redirect (plus TLS/rule hosts) on it, so data-plane renders with globalBaseDomain unset no longer template `<no value>`. Added a data-plane regression test.

File: tests/chart_tests/test_dual_host_ingress.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+def _tls_hosts(ingress):
+    hosts = []
+    for entry in ingress["spec"].get("tls", []):
+        hosts.extend(entry.get("hosts", []))
+    return hosts
+
+
+@pytest.mark.parametrize("kube_version", supported_k8s_versions)
+@pytest.mark.parametrize("ingress_file", list(INGRESS_HOSTS))
```
Comment (danielhoherd) [inline tests/chart_tests/test_dual_host_ingress.py:71]: Nit: why is this called `ingress_file` ? Is there rally a file? It looks like there isn't.

**Review by danielhoherd — APPROVED** [review]
Seems good to me
